geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/cluster/RedisMemberInfoRetrievalFunction.java

```diff
@@ -53,49 +47,31 @@ private BucketInfoRetrievalFunction(String address, int redisPort) {
   }
 
   public static void register(String address, int redisPort) {
-    FunctionService.registerFunction(new BucketInfoRetrievalFunction(address, redisPort));
+    FunctionService.registerFunction(new RedisMemberInfoRetrievalFunction(address, redisPort));
   }
 
   @Override
   public void execute(FunctionContext<Void> context) {
-    Region<RedisKey, ByteArrayWrapper> region =
-        context.getCache().getRegion(RegionProvider.REDIS_DATA_REGION);
-
     String memberId =
         context.getCache().getDistributedSystem().getDistributedMember().getUniqueId();
 
-    MemberBuckets mb = new MemberBuckets(memberId, hostAddress, redisPort,
-        getLocalPrimaryBucketIds(region));
-    context.getResultSender().lastResult(mb);
+    context.getResultSender().lastResult(new RedisMemberInfo(memberId, hostAddress, redisPort));
   }
 
   @Override
   public String getId() {
     return ID;
   }
 
-  @SuppressWarnings("unchecked")
-  public static Set<Integer> getLocalPrimaryBucketIds(Region region) {
-    PartitionedRegion pr = (PartitionedRegion) region;
-    if (pr.getDataStore() != null) {
-      return new HashSet<>(pr.getDataStore().getAllLocalPrimaryBucketIds());
-    } else {
-      return Collections.EMPTY_SET;
-    }
-  }
-
-  public static class MemberBuckets implements Serializable {
+  public static class RedisMemberInfo implements Serializable {
```

Comment:
This should probably be BasicSerializable or even DataSerializableFixedID, rather than a java serializable.

Reply:
Now implements `DataSerializableFixedID`